clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy

```diff
@@ -96,51 +97,81 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
         errList = AzureDeploymentOperation.checkDeploymentOperationStatus(task, BASE_PHASE, description.credentials, resourceGroupName, deployment.name)
       } else {
         // We are attempting to create a new application gateway
-        description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, virtualNetworkName, description.region)
+        if (!description.useDefaultVnet) {
```

Comment:
is the idea that if *useDefaultVnet* is false then we will try to use the vnet specified in the description and if it's true, then we will create a new vNet?